### ISSUE TITLE
[Feature Request] Make bullets slow down in liquids

### ISSUE THREAD

**Issue by Chris6ix** [VulpesStella/MrCrayfishGunMod#11]
It would be great if there was some kind of drag applied to the bullets when they hit/go through liquids so that you can't shoot (or at least not as easily shoot at) people/mobs that are submerged in a liquid or by you waiting in one for people to swim across, for example.

Maybe this could be a set amount of drag per "block of liquid", so that it's still possible to for example shoot someone that is 1 or 2 blocks submerged, but not someone that is lets say 5 blocks deep.

These liquids would obviously be water and lava, but it would be amazing if this would also work with modded liquids such as acid or oil if they are registered as a liquid.

**Comment by VulpesStella** [comment]
What I could do was add gravity & resistance for projectiles in the water. Launched grenades and missiles worked fine. Bullets were too fast and I could not make sure whether it was working.

**Comment by Chris6ix** [comment]
Can you share that build? I could test it and tell you if it works.

**Comment by VulpesStella** [comment]
> Can you share that build? I could test it and tell you if it works.

Wait a minute :D

**Comment by VulpesStella** [comment]
[UnzipMe.zip](https://github.com/MikhailTapio/MrCrayfishGunMod/files/14662396/UnzipMe.zip)

**Comment by Chris6ix** [comment]
Thank you.
I have tested it and yes, normal bullets are not affected at all, I was able to sit underwater and snipe all the fish 20+ blocks away without having to adjust my aim or anything like that.
It also seems to have something to do with the individual liquids, because the grenade did not get slowed down in any liquid I tested that is not water
In Water, both the missile and grenade were slowed down
In Lava, only the missile was slowed down
In Crude Oil from the Create Diesel Generators mod, only the missile was slowed down
In Bio Diesel from the Create Diesel Generators mod, only the missile was slowed down

Also, there is now a bug that the missile doesn't have any drop off at all outside of liquids.
